pkg/auth/providers/common/config.go

```diff
@@ -0,0 +1,24 @@
+package common
+
+const (
+	// CleanupAnnotation exists to prevent admins from running the cleanup routine in two scenarios:
+	// 1. When the provider has not been enabled or deliberately disabled, and thus does not need cleanup.
+	// 2. When the value of the annotation is 'user-locked', set manually by admins in advance.
+	// Rancher will run cleanup only if the provider becomes disabled,
+	// and the annotation's value is 'unlocked'.
+	CleanupAnnotation = "management.cattle.io/auth-provider-cleanup"
+
+	CleanupUnlocked      = "unlocked"
+	CleanupUserLocked    = "user-locked"
+	CleanupRancherLocked = "rancher-locked"
+)
+
+// ResetAuthConfig takes an Auth Config as a map and deletes all entries except those with basic metadata fields.
+func ResetAuthConfig(cfg map[string]any) {
```

Comment:
If this is now only being used in one place why create its own package for it? Just put it in the package where it is being used. The consts could also be moved back where they came from.